medicles/templates/medicles/detail.html

```diff
@@ -266,92 +376,219 @@ <h4 id="tagHeader" title="Tag me" class="w3-text-teal w3-left w3-text-opacity">
+        document.getElementById("articleAbstract").innerHTML = document.getElementById("articleAbstractCopy").innerHTML
+        document.getElementById("annotationInfo-division").hidden = true
+        document.getElementById("article-division").setAttribute("style", "width:fit-content");
+    }
+
+
+    <!-- Highlight text for annotation-->
+
+    document.getElementById('articleAbstract').addEventListener('mouseup', function (e) {
+
+        var txt = this.innerText;
+        var selection = window.getSelection();
+        var start = selection.anchorOffset;
+        var end = selection.focusOffset;
+
+        annotationItemToBeShownInDialog = selection.anchorNode.wholeText
+
+        var conflictedAnnotation = false
+        let offsetToStart = 0
+
+        for (let annotationText of annotationList) {
+            if (annotationText.target.text.trim() === selection.focusNode.data.trim()) {
+                conflictedAnnotation = true
+            }
+        }
+        if (conflictedAnnotation) {
+
+            let start = parseInt(txt.trim().indexOf(selection.focusNode.data.trim())) + parseInt(selection.extentOffset) - 1
+            document.getElementById("annotationStartIndex").value = start
+            document.getElementById("annotationEndIndex").value = start + selection.toString().length
+        } else {
+            if (start > end) {
+                let temp = start
+                start = end
+                end = temp
+            }
+            offsetToStart = txt.trim().indexOf(selection.focusNode.data.trim()) - 1
+            document.getElementById("annotationStartIndex").value = start + parseInt(offsetToStart)
+            document.getElementById("annotationEndIndex").value = end + parseInt(offsetToStart)
         }
 
-        jQuery(function () {
-            $("#id_tag_key").on('keyup', function () {
-                let value = $(this).val();
-                $.ajax({
-                    url: "{% url 'medicles:ajax_load_tag' %}",
-                    data: {
-                        'tag_query': value
-                    },
-                    dataType: 'json',
-                    success: function (data) {
-                        let tags = data.tags;
-                        $("#id_tag_key").autocomplete({
-                            source: tags,
-                            minLength: 2
-                        });
-                    }
-                });
-            });
+        document.getElementById("textToBeAnnotated").value = selection.toString()
+        document.getElementById("annotationField").disabled = false;
+        document.getElementById("annotationBT").disabled = false
 
+    });
 
-        });
-    </script>
 
-    <script type="text/javascript">
+    <!-- Cancel text for annotation-->
 
-        $('#favourite').click(function () {
+    function cancelAnnotationProcess() {
 
-            $.post('{% url "medicles:favourite_article" article.article_id %}');
```

Comment:
this script is also for the favourite feature which binds the favourite button to the view

Reply:
I added this part again